### PR TITLE
implemented basic auth

### PR DESCRIPTION
**Pull request by ckieverentineas**

### CODE REVIEW

File: controllers/auth.mjs [hunk 1]
```diff
@@ -1,17 +1,39 @@
...
 
 router.post('/signup', async (req, res) => {
   const { email, password } = req.body;
-  const user = await db('Accounts').where({ email });
+  const [user] = await db('Accounts').where({ email });
   if (!user) {
-    await db('Accounts').insert({ email, password, account_created: Date.now() });
-    res.redirect('/signin');
+    const hash = bcrypt.hashSync(password, 8);
+    await db('Accounts').insert({ email, password: hash, account_created: Date.now() });
```
Comment (MurkyMeow) [inline controllers/auth.mjs]: You should put try-catch here and response with Internal Server Error (500) if the server couldn't access the database. Waiting for connection to timeout feels very wrong =).

File: controllers/auth.mjs [hunk 2]
```diff
@@ -1,17 +1,39 @@
...
+    res.json({ status:"created account" });
+  } else {
+    res.status(400).json({ message: 'Fail to create account' });
   }
 });
 
-router.post('/signin', (req, res) => {});
+router.post('/signin', async (req, res) => {
+  const { email, password } = req.body;
+  const [user] = await db('Accounts').where({ email });
```
Comment (MurkyMeow) [inline controllers/auth.mjs]: This is also requires a try-catch.

File: controllers/auth.mjs [hunk 3]
```diff
@@ -1,17 +1,39 @@
-import { Router } from 'express';
+//import { Router } from 'express';
+import express from 'express';
+import bcrypt from 'bcryptjs';
+import jwt from 'jsonwebtoken'
```
Comment (MurkyMeow) [inline controllers/auth.mjs]: There are a lot of mistakes like this. Do you have eslint installed?
```suggestion
import jwt from 'jsonwebtoken';
```

**Comment by MurkyMeow** [comment]
Or even better -- add a separate error handler.
http://expressjs.com/en/guide/error-handling.html

File: controllers/auth.mjs (hunk 1 above)
Comment (ckieverentineas) [reply controllers/auth.mjs]: How doing it?

Comment (MurkyMeow) [reply controllers/auth.mjs]: Nevermind, that external error handler solves that.

**Pull request merged**